src/proof/PGTransformationAlgorithms.cc

```diff
@@ -1057,5 +1001,122 @@ void ProofGraph::proofPostStructuralHashing()
 	}
 	///////////////////////////////////////////////////////////////////////////////
 }
-#endif
+
+
+namespace{
+class LightVars {
+public:
+
+    template<typename TIt>
+    LightVars(TIt begin, TIt end) {
+        lights.insert(begin, end);
+    }
+
+    bool isLight(Var v) const { return lights.find(v) != lights.end(); }
+
+private:
+    std::unordered_set<Var> lights;
+};
+
+class LiftingVarsRuleHandler {
+public:
+    LiftingVarsRuleHandler(ProofGraph & pg, LightVars const & lv) : proofGraph(pg), lightVars(lv) {}
+
+    short operator()(RuleContext & ra1, RuleContext & ra2);
+
+private:
+    ProofGraph & proofGraph;
+    LightVars const & lightVars;
+};
+
+short LiftingVarsRuleHandler::operator()(RuleContext &ra1, RuleContext &ra2) {
+    // Return 0 for no application, 1 for applying rule 1, 2 for applying rule 2
```

Comment:
Consider returning an enum with nicer names?

Reply:
Good idea!

Reply:
I think I managed to refactor this to use enum